src/imap/select_folder.rs

```diff
@@ -119,7 +120,15 @@ impl Imap {
                 Err(Error::NoSession)
             }
         } else {
-            Ok(())
+            Ok(NewlySelected::No)
         }
     }
 }
+#[derive(PartialEq)]
```

Comment:
Derive `Debug`, `Copy` and `Eq` too, they are generally useful. I thought [`missing_debug_implementations`](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#missing-debug-implementations) lint was enabled, apparently it isn't.